### DISCUSSION TITLE
What's the difference between location and directory settings?

### DISCUSSION THREAD

**Discussion by GiovanniSalmeri** [datenstrom/community#409] [Ask a question] [answered]
I am afraid my question is silly, but I couldn't find an answer in the docs or in past issues: what's the difference between for example `DownloadLocation` and `DownloadDir`? There are several similar twin settings. In the default settings I see only an additional slash in the first case and also after reading the code I am a bit confused. I ask this because I am writing a new extension and I would like to follow as exactly as possible the patterns used elsewhere. Thank you!

**Comment by schulle4u** [comment, marked as answer]
The `*Location` settings  are used for web-accessible paths, whereas the `*Dir` settings are directories relative to the Yellow directory. As a comprehensible example have a look at `ExtensionLocation` and `ExtensionDir`. The first points to the web-accessible virtual storage for extension files and can be used e.g. for linking CSS ressources (translates to domain + `/media/extensions`), the second is the real system path for the extensions (YellowDir + `system/media/extensions`). 

HTH
Steffen

**Reply by markseuffert** [reply]
There was an API update in April 2022. We simplified extensions so that there will be mainly `Location` settings. If your extension needs the directory from a location you can get it programmatically via the API. Let's take for example the setting `coreImageLocation` and you would like to get the `coreImageDirectory`, now you can do this:

`$path = $this->yellow->lookup->findMediaDirectory("coreImageLocation");`

Some directories have no `Location` settings, then you can do this:

`$path = $this->yellow->system->get("coreContentDirectory");`

Some special cases have `Location` and `Directory` settings, as Steffen explained above.

For more code examples see [API for developers](https://datenstrom.se/yellow/help/api-for-developers) and [image extension](https://github.com/annaesvensson/yellow-image). Hope you find this useful.

**Comment by GiovanniSalmeri** [comment]
Alles klar, thank you very much for the quick answer!
